semgrep-core/Parsing/other/yaml_to_generic.ml

```diff
@@ -172,15 +176,135 @@ let parse env parser : A.expr =
+          (ellipses_space ^ sgrep_ellipses)::converted_ellipses, rest )
+        else [], ellipses
+      end
+  in
+  convert_ellipses ellipses
+
+let split_whitespace line =
+  let line_len = String.length line in
+  let rec read_string i =
+    if i = line_len then line
+    else begin
+      match line.[i] with
+      | ' ' | '\t' | '-' -> read_string (i + 1)
+      | _ -> substring line 0 i
+    end
+  in
+  let whitespace = read_string 0 in
+  if String.contains whitespace '-' then
+    String.index whitespace '-', whitespace else
+    String.length whitespace, whitespace
+
+(* Split a line by the ellipses *)
+let split_on_ellipses line =
+  let rec split line i ellipses_start num_dots =
+    let line_length = String.length line in
+    if i = line_length then [line] else
+      begin
+        match line.[i] with
+        | '.' -> begin
+            match num_dots with
+            | 0 -> split line (i + 1) i 1
+            | 1 -> split line (i + 1) ellipses_start 2
+            | 2 -> (substring line 0 ellipses_start)::(split (substring line (i + 1) line_length) 0 0 0)
+            | _ -> raise ImpossibleDots
+          end
+        | _ -> split line (i + 1) 0 0
+      end
+  in
+  let pieces = split line 0 0 0 in
+  pieces
+
+let rec exit_context whitespace_len context =
+  match context with
+  | [] -> []
+  | (prev_len, _)::rest -> begin
+      if prev_len > whitespace_len then exit_context whitespace_len rest
+      else context
+    end
+
+
+(***** Preprocess the yaml *****)
+
+let preprocess_yaml str =
+  let lines = String.split_on_char '\n' str in
+  (* Main processing function *)
+  let rec process_lines lines context ellipses =
+    match lines with
+    | [] -> []
+    | line::rest -> begin
+        let whitespace_len, whitespace = split_whitespace line in
+        let context', line', ellipses' =
+          match String.trim line with (* This uses line to check for "..." vs "- ..." *)
+          | "..." -> begin
+              match last_same_whitespace whitespace context with
+              | Some ws -> context, [ws ^ sgrep_ellipses], ellipses
+              | None -> context, [], (whitespace_len, whitespace)::ellipses
+            end
+          | _ ->
+              let converted_ellipses, ellipses' = convert_leftover_ellipses (whitespace_len, whitespace) ellipses in
+              (whitespace_len, whitespace)::(exit_context whitespace_len context),
+              converted_ellipses @ [String.concat sgrep_ellipses_inline (split_on_ellipses line)],
+              ellipses'
+        in
+        line'@(process_lines rest context' ellipses')
+      end
+  in
+  String.concat "\n" (process_lines lines [] [])
+
+
 (*****************************************************************************)
 (* Entry points *)
 (*****************************************************************************)
 
 let program file =
-  let str = Common.read_file file in
+  let str = preprocess_yaml (Common.read_file file) in
```

Comment:
you should not need that for the target file. Only the pattern can contain ellipsis.